pkg/provider/provider.go

```diff
@@ -378,7 +379,104 @@ func (k *kubeProvider) Invoke(ctx context.Context,
 
 	// Always fail.
 	tok := req.GetTok()
-	return nil, fmt.Errorf("Unknown Invoke type '%s'", tok)
+	label := fmt.Sprintf("%s.Invoke(%s)", k.label(), tok)
+	args, err := plugin.UnmarshalProperties(
+		req.GetArgs(), plugin.MarshalOptions{Label: label, KeepUnknowns: true})
+	if err != nil {
+		return nil, pkgerrors.Wrapf(err, "failed to unmarshal %v args during an Invoke call", tok)
+	}
+
+	switch tok {
+	case invokeParseYaml:
+		// invokeParseYaml is used by the YAML and Helm SDKs to load YAML from a variety of input sources,
+		// parse this input into structs, and then return the results in a data structure that can be
+		// turned into Pulumi resources.
+		//
+		// Input can be one of the following types:
+		// 1. File path
+		// 2. File path glob (e.g., *.yaml)
+		// 3. URL
+		// 4. Literal YAML string
+		//
+		// Output is of the following form:
+		// { name: { path: [Object] } }
+
+		// input: []{ name: string, type: "yaml" | "path", value: string }
+		var input []resource.PropertyValue
+		if args["input"].HasValue() {
+			input = args["input"].ArrayValue()
+		}
+
+		// result: { name: { path: []interface{} } }
+		result := map[string]map[string]interface{}{}
+
+		for _, v := range input {
```

Comment:
We should be leveraging go routines here. This will help parallelize parsing if multiple inputs are provided (e.g. file path glob)